paperqa/agents/search.py

```diff
@@ -525,9 +556,13 @@ async def process_file(
                 },
                 document=tmp_docs,
             )
-            # Save so we can resume the build without rebuilding this file if a
-            # separate process_file invocation leads to a segfault or crash
-            await search_index.save_index()
+
+            processed += 1
+            if processed == settings.agent.index.concurrency:
```

Comment:
let's make a separate index setting `batch_size` or something like this, and default it to `1`